README.md

```diff
@@ -1,6 +1,89 @@
-# JanusGraph Python driver
+# JanusGraph-Python
 
-## License
+## Python Client drivers for [JanusGraph](http://janusgraph.org)
+
+JanusGraph-Python is the Python drivers for connecting to JanusGraph. 
+It extends Apache TinkerPop™'s [Gremlin-Python](http://tinkerpop.apache.org/docs/current/reference/#_gremlin_python) 
+as its core dependency with additional support for JanusGraph-specific types and predicates.
+
+[![Codacy Badge](https://api.codacy.com/project/badge/Grade/a643a483556740c1b87ba29e160d37b6)](https://www.codacy.com/app/JanusGraph/janusgraph-python?utm_source=github.com&utm_medium=referral&utm_content=JanusGraph/janusgraph-python&utm_campaign=Badge_Grade)
+
+### Pre-Requisites
+
+The library has been tested with following Python versions:
+
+-   [Python 3.4](https://www.python.org/downloads/release/python-340/).
+-   [Python 3.5](https://www.python.org/downloads/release/python-350/).
+-   [Python 3.6](https://www.python.org/downloads/release/python-360/).
+
+Once the required Python version is installed on system, please note the version number and the Python added 
+to Path. Make a note of the number, as that will be required during building of Library (on UNIX)
+Try running either of following commands on CLI, and make a note of the version which works:
+
+-   python3 --version
+-   python3.4 --version
+-   python3.5 --version
+-   python3.6 --version
+
+**NOTE**: The above checks aren't needed if installing the drivers from PyPi
+
+### Building Library
+
+You can build the library yourself to test out the functionality of library. Refer to 
+[Building docs](BUILDING.md) for documentation on how to build the library.
+
+### Installing the Library
+
+-   Though Installation of library is built as a functionality of the automated build scripts provided,
+      but if needed, user can follow the bellow instructions to install the library to environment of their choice.
+
+-   To install library using Pip (It is not yet hosted):
+
+    ```bash
+    # X is version number of JanusGraph Python client supported based on JanusGraph version chosen.
+    pip install janusgraph_python=X
+    ```
+    
+-   To install library from tarball, once it is built:
+
+    ```bash
+    pip install target/dist/janusgraph_python/dist/janusgraph_python-X.tar.gz
+    # X is version number of JanusGraph Python client supported based on JanusGraph version chosen.
+    ```
+
+### Compatibility Matrix
+
+| JanusGraph Version | Client Version |
```

Comment:
Not that important but I would note the client version first and then the JanusGraph version as a single version of the driver can be compatible with multiple JanusGraph versions. There were no changes relevant to the drivers between 0.4 and 0.5 for example.
It would also be good to stay consistent with how the versions are listed. You list 0.2.x for all 0.2 versions, but then list 0.3.0 and 0.3.1 explicitly.

(See [the same table for JanusGraph.Net](https://github.com/JanusGraph/janusgraph-dotnet#version-compatibility) for example. Maybe we stay consistent between the different languages?)

Reply:
Makes sense, I updated this section to reflect like you have in .NET. Consistency should be the key :-)
But. makes me wonder, what if tomorrow we bring in a new feature, let's say additional Geotypes, and it goes into version 0.1.1 of client, so we will add 2 rows, 0.1.0 and 0.1.1 in Client column for same JanusGraph? Just a question :-)